extensions/ql-vscode/src/pure/files.ts

```diff
@@ -28,3 +28,25 @@ export async function gatherQlFiles(paths: string[]): Promise<[string[], boolean
   }
   return [Array.from(gatheredUris), dirFound];
 }
+
+/**
+ * Lists the names of directories inside the given path.
+ * @param path The path to the directory to read.
+ * @returns the names of the directories inside the given path.
+ */
+export async function getDirectoryNamesInsidePath(path: string): Promise<string[]> {
```

Comment:
This seems like something we can create a test for. Do you think it would make sense to add a test for this in `files.test.ts`?

Reply:
I've added some very simple tests - we could do with something a bit more advanced/extensive but I think we might want to tidy up the test data dirs first.